src/schedule.rs

```diff
@@ -1,18 +1,22 @@
-use crate::error::{Error, ErrorKind};
+
 use chrono::offset::TimeZone;
-use chrono::{DateTime, Datelike, Duration, Timelike, Utc};
-use nom::{types::CompleteStr as Input, *};
-use std::collections::BTreeSet;
+use chrono::{DateTime, Datelike, Utc};
 use std::collections::Bound::{Included, Unbounded};
 use std::fmt::{Display, Formatter, Result as FmtResult};
-use std::iter::{self, Iterator};
-use std::str::{self, FromStr};
 
 use crate::time_unit::*;
+use crate::ordinal::*;
+use crate::queries::*;
+
+impl From<Schedule> for String {
+    fn from(schedule: Schedule) -> String {
+        schedule.source.unwrap()
+    }
+}
 
 #[derive(Clone, Debug)]
 pub struct Schedule {
-    source: Option<String>,
+    pub(crate) source: Option<String>, //TODO: remove pub
```

Comment:
Ordinarily I'd suggest adding a method for this setting value instead of making a single field public, but since you're getting ready to restructure this anyway it's fine for now.

Reply:
Yeah, #71 would fix this. I already wrote the code for that issue, but I'm not sure if I should expose ScheduleFields or not.